src/ThreadMaster.java

```diff
@@ -0,0 +1,17 @@
+public class ThreadMaster extends Thread{
+    static int counter = 0;
```

Comment:
this is not a shared counter, but rather a field of an instance of ThreadMaster. So, the values will always be "correct" since there is no shared resource (memory).